spec/models/user_spec.rb

```diff
@@ -1,5 +1,32 @@
 require 'rails_helper'
 
 RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+  describe 'validates' do
+    subject { User.new name: 'Usman Suleiman' }
+
+    before { subject.save }
+
+    it 'should have a name' do
+      subject.name = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'should have an alpha name' do
+      subject.name = ' '
+      expect(subject).to_not be_valid
+    end
+
+    it 'should have a postive integer posts counter' do
+      subject.posts_counter = -1
+      expect(subject).to_not be_valid
+    end
```

Comment:
- [ ] This test should be changed because post_counter will not only be positive integer but it will accept integers which is greater than zero

Reply:
@DeliceLydia